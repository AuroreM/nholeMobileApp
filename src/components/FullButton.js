// @flow
import React, { Component } from 'react';
import { StyleSheet, View, ActivityIndicator, Text } from 'react-native';
import Button from './Button';

class FullButton extends Component<Props, void> {
  render() {
    const styles = getStyles();
    const { title, secondaryButton, ...rest } = this.props;
    return (
      <Button {...rest}>
        <View style={[styles.button, secondaryButton && styles.secondaryButton]}>
          <Text style={[styles.buttonText, secondaryButton && styles.secondaryButtonText]}>{title}</Text>
        </View>
      </Button>
    );
  }
}

type Props = {
  title: string,
  secondaryButton?: boolean,
};

const getStyles = () =>
  StyleSheet.create({
    button: {
      backgroundColor: 'rgb(64,64,64)',
      borderRadius: 5,
      width: 200,
      height: 40,
      alignItems: 'center',
      justifyContent: 'center',
    },
    secondaryButton: {
      backgroundColor: 'transparent',
      borderWidth: 0,
    },
    buttonText: {
      color: 'rgb(30,144,255)',
      fontSize: 18,
    },
    secondaryButtonText: {
      color: 'rgb(64,64,64)',
      fontSize: 16,
    },
  });
export default FullButton;
