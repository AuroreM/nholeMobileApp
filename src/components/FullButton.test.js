import React from 'react';
import 'react-native';
import FullButton from './FullButton';
import renderer from 'react-test-renderer';

describe('<FullButton />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<FullButton title="Envoyer" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders correctly when it is a secondary Button', () => {
    const tree = renderer.create(<FullButton title="Envoyer" secondaryButton />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
