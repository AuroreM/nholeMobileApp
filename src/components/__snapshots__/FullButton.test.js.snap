// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FullButton /> renders correctly 1`] = `
<View
  activeOpacity={1}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "opacity": 0.9,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "rgb(64,64,64)",
          "borderRadius": 5,
          "height": 40,
          "justifyContent": "center",
          "width": 200,
        },
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "rgb(30,144,255)",
            "fontSize": 18,
          },
          undefined,
        ]
      }
    >
      Envoyer
    </Text>
  </View>
</View>
`;

exports[`<FullButton /> renders correctly when it is a secondary Button 1`] = `
<View
  activeOpacity={1}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "opacity": 0.9,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "rgb(64,64,64)",
          "borderRadius": 5,
          "height": 40,
          "justifyContent": "center",
          "width": 200,
        },
        Object {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "rgb(30,144,255)",
            "fontSize": 18,
          },
          Object {
            "color": "rgb(64,64,64)",
            "fontSize": 16,
          },
        ]
      }
    >
      Envoyer
    </Text>
  </View>
</View>
`;
